### PR TITLE
feat(sdk, gate, cli):  upload protocol poc  uploading wasm file for `WasmEdge Runtime` for single replica mode

### PR DESCRIPTION
**Pull request by destifo**
Upload protocol for wasm files and atrifacts for `WasmEdge Runtime` for single replica mode

#### Motivation and context

- Upload WasmEdge Runtime artifacts during typegraph deploy
- Access and load WasmEdge Runtime artifacts from the local file system from typegate

#### Migration notes

*No Migrations Needed*

### Checklist

- [x] The change come with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

**Review by Natoandro — CHANGES_REQUESTED** [review]
Great work :muscle: 
A few comments...

File: meta-cli/src/deploy/actors/loader.rs [hunk 1]
```diff
@@ -125,6 +125,10 @@ pub enum ReloadReason {
     DependencyChanged(PathBuf),
 }
 
+pub struct ProcFlag {
+    pub include: Vec<String>,
+}
+
```
Comment (afmika) [inline meta-cli/src/deploy/actors/loader.rs:131]: This is unused, I rm'd this on main.

File: sample/example/another.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Policy, t, typegraph } from "../../typegraph/node/sdk/dist/index.js";
+import { DenoRuntime } from "../../typegraph/node/sdk/dist/runtimes/deno.js";
+
+await typegraph({
```
Comment (afmika) [inline sample/example/another.ts:4]: Same as above

**Review by afmika — COMMENTED** [review]
Nice 👍

File: meta-cli/src/deploy/actors/loader.rs (hunk 1 above)
Comment (destifo) [reply meta-cli/src/deploy/actors/loader.rs:131]: Done.

File: typegraph/core/src/utils/postprocess/wasmedge_rt.rs [hunk 3]
```diff
@@ -28,8 +25,14 @@ impl PostProcessor for WasmedgeProcessor {
...
 
                 let wasi_path = fs_host::make_absolute(&PathBuf::from(path))?;
-                let bytes = read_file(&wasi_path.display().to_string())?;
-                mat_data.wasm = encode_bytes_to_base_64(bytes).map_err(|e| e.to_string())?;
+                let file_name = path.split('/').last().unwrap();
+                let artifact_hash = mat_data.artifact_hash.clone();
+                let wasm_path = format!(
+                    "tmp/metatype_artifacts/{}/files/{}.{}",
+                    tg_name, file_name, artifact_hash
+                );
```
[the remark this answers is not in the document]
Comment (destifo) [reply typegraph/core/src/utils/postprocess/wasmedge_rt.rs:33]: yeah, it is. Just need to add `artifact_hash` field to `WasiInput` or just modify the filename in form of `${file_name}.${artifact_hash}` then create path on typegate. Now I think about it, it gives the typegate the freedom to choose where to store the artifacts rather the tg sdks which don't really need to care about where they are placed. on it.

File: typegate/engine/src/runtimes/wasmedge.rs [hunk 4]
```diff
@@ -54,9 +56,16 @@ fn param_cast(out: &str, res: &mut Vec<Box<dyn Any + Send + Sync>>) -> Result<St
...
+    let wasm_relative_path = PathBuf::from(input.wasm);
+
+    let wasm_absolute_path = match env::current_dir() {
+        Ok(cwd) => cwd.join(wasm_relative_path),
+        Err(e) => return Err(anyhow::anyhow!(e)),
+    };
+
+    let wasm_binary = common::archive::encode_to_base_64(&wasm_absolute_path).unwrap();
+
+    let bytes = general_purpose::STANDARD.decode(wasm_binary).unwrap();
```
[the remark this answers is not in the document]
Comment (destifo) [reply typegate/engine/src/runtimes/wasmedge.rs:68]: Done.

File: typegraph/core/src/utils/postprocess/wasmedge_rt.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (destifo) [reply typegraph/core/src/utils/postprocess/wasmedge_rt.rs:33]: Done.

File: typegraph/node/sdk/src/tg_deploy.ts [hunk 5]
```diff
@@ -51,14 +61,60 @@ export async function tgDeploy(
...
-    serialized,
+
+  const result = {
+    serialized: tgJson,
     typegate: await handleResponse(response),
   };
+
+  const suffix = `${typegraph.name}/get-upload-url`;
+  const getUploadUrl = new URL(suffix, baseUrl);
+  for (let [fileHash, filePath] of ref_files) {
```
[the remark this answers is not in the document]
Comment (destifo) [reply typegraph/node/sdk/src/tg_deploy.ts:63]: it's done @Natoandro.

File: typegate/src/runtimes/wasmedge.ts [hunk 6]
```diff
@@ -40,7 +40,8 @@ export class WasmEdgeRuntime extends Runtime {
         await native.wasmedge_wasi(
           {
             func: func as string,
-            wasm: wasm as string,
+            wasm:
+              `tmp/metatype_artifacts/${tg_name as string}/files/${wasm as string}.${artifact_hash as string}`,
```
Comment (Yohe-Am) [inline typegate/src/runtimes/wasmedge.ts:44]: thought: noticed you were reading from `config.tmp_dir` down below. Should it not be the same here?

**Review by Yohe-Am — APPROVED** [review]
Good stuff. Can't wait till we're at the end of the wasm milestone and all this work comes together.

File: typegate/src/runtimes/wasmedge.ts (hunk 6 above)
Comment (destifo) [reply typegate/src/runtimes/wasmedge.ts:44]: Good eye! didn't make changes after I changed where the path resolves. I will correct it.

Comment (destifo) [reply typegate/src/runtimes/wasmedge.ts]: Done.

**Review by Natoandro — DISMISSED** [review]

**Review by Natoandro — APPROVED** [review]

**Pull request merged**